app.spec.js

```diff
@@ -0,0 +1,401 @@
+	});
+
+	it('can fetch all users', async function () {
+		await models.dataReset();
+
+		let u1 = new User(fixture);
+		await User.insert(u1);
+
+		let u2 = new User(fixture);
+		u2.id = u2.id + 'aaa';
+		await User.insert(u2);
+
+		let headers = {Authorization: 'abc'};
+		let resp = await client.request('GET', '/v1/users', null, headers);
+		let body = JSON.parse(resp.body);
+		let ids = new Set([u1.id, u2.id]);
+		assert(resp.statusCode === 200);
+		assert(body.length === 2);
+
+		for (let item of body) {
+			ids.delete(item.id);
+		}
+		assert(ids.size === 0);
+	});
+
+	it('returns 500 on error fetching all users', async function () {
+		// error case
+		let save = User.getAll;
+		User.getAll = () => new Error();
+		let headers = {Authorization: 'abc'};
+		let resp = await client.request('GET', '/v1/users', null, headers);
+		assert(resp.statusCode === 500);
+
+		// restore
+		User.getAll = save;
+	});
+
+	it('can fetch a single user', async function () {
+		let u = new User(fixture);
+		let inserted = await User.insert(u);
+
+		let id = u.id;
+		let headers = {Authorization: 'abc'};
+		let resp = await client.request('GET', `/v1/users/${id}`, null, headers);
+		let body = JSON.parse(resp.body);
+		assert(resp.statusCode === 200);
+		for (let k of Object.keys(inserted)) {
+			assert(inserted[k] === body[k]);
+		}
+	});
+
+	it('returns a 404 if the user does not exist', async function () {
+		let headers = {Authorization: 'abc'};
+		let resp = await client.request('GET', `/v1/users/nonexistent`, null, headers);
+		let body = JSON.parse(resp.body);
+		assert(resp.statusCode === 404);
+		assert(body.error.includes('no such user'));
+	});
+
+	it('returns 500 on error fetching a single user', async function () {
+		let save = User.fetch;
+		User.fetch = () => new Error();
+
+		let headers = {Authorization: 'abc'};
+		let resp = await client.request('GET', `/v1/users/some-id`, null, headers);
+		assert(resp.statusCode === 500);
+
+		// restore
+		User.fetch = save;
+	});
+
+	it('can post a new user', async function () {
+		await models.dataReset();
+
+		let user = new User(fixture);
+		let headers = {Authorization: 'abc'};
+		let resp = await client.request('POST', '/v1/users', user, headers);
+		let inserted = JSON.parse(resp.body);
+		let iso8601Pattern = /^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(.[0-9]+)?Z$/;
+
+		assert(resp.statusCode === 200);
+		assert(inserted.id === user.id);
+		assert(inserted.name === user.name);
+		assert(inserted.dob === user.dob);
+		assert(inserted.address === user.address);
+		assert(inserted.description === user.description);
+		assert(iso8601Pattern.test(inserted.createdAt));
+		assert(iso8601Pattern.test(inserted.updatedAt));
+
+		let fetched = User.fetch(inserted.id);
+		for (let k of Object.keys(fetched)) {
+			assert(fetched[k] === inserted[k]);
+		}
+	});
+
+	it('rejects invalid data on user post', async function () {
```

Comment:
yeah, that's fair.. I guess my mind grouped them together initially because of the similarity but splitting out def doesn't hurt here, and pretty simple to do.